DataLoader/src/data-loader.ts

```diff
@@ -0,0 +1,85 @@
+import L from 'leaflet';
+import './data-loader.css';
+
+interface DataLoaderOptions extends L.ControlOptions {
+  // TODO: Get the real list of options from Seneca
+  query: any; // Define the structure of your query object
+  senecaPostFunction: Function; // Function to handle POST requests
+  entityNames: string[]; // List of entities to load
+  config: any; // Additional configuration if needed
+}
+
+interface DataLoaderData {
+  [key: string]: any[]; // Replace `any[]` with more specific types if available
+}
+
+
+const DataLoader = L.Control.extend({
+  options: {
+    position: 'topright',
+    entityNames: ['building', 'level', 'room', 'geofence', 'asset'], // Default entities
+  } as DataLoaderOptions,
+
+  _data: {} as DataLoaderData,
+  _isLoading: false,
+  _isLoaded: false,
+
+  initialize: function (options: DataLoaderOptions) {
+    L.Util.setOptions(this, options);
+    this._data = {};
+  },
+
+  loadData: async function () {
+    if (this._isLoading) {
```

Comment:
remove this checking code with the timeouts, but keep the isLoading status flag - that should be public so remove the _

Reply:
Done!